core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java

```diff
@@ -156,17 +156,46 @@ public void handleUnsubscribe(Rbox.UnsubscribeRequest request, StreamObserver<Em
 
         @Override
         public void handleUpdate(Rbox.UpdateMessage request, StreamObserver<Empty> responseObserver) {
-            // TODO: Call ChangeReceiver
+            logger.log(Level.INFO, "Handling update...");
+
+            // Call ChangeReceiver
+            // TODO: Cannot reference receiveChange from static context?
+            ChangeReceiver.receiveChange(request.getRemoteChange());
+
+            sendToReplicaHolders(request.getTargetObject(), request.getRemoteChange());
+
+            // TODO: Do we need to send a response?
         }
     }
 
-    void handleQueryResult(GameObjectUUID primaryObjectUUID, List<HolderInfo> interestedObjects) {
+    void handleQueryResult(GameObjectUUID primaryObjectUUID, List<edu.rice.rbox.Replication.HolderInfo> interestedObjects) {
         // TODO: Subscribe & Unsubscribe when necessary
     }
 
-    /* Functions for Object Storgae */
+    /* Functions for Object Storage */
     void updatePrimary(RemoteChange change) {
-        // TODO: Send change to primary holder
+        // Send the response
+        logger.log(Level.INFO, "Sending change to primary...");
+
+        // Get primary
+        // TODO: What if we're the primary?
```

Comment:
This is handled by Object Storage, so it is only called when we're the replica holder